language/proto/generate.go

```diff
@@ -41,13 +41,32 @@ func (_ *protoLang) GenerateRules(args language.GenerateArgs) language.GenerateR
 			regularProtoFiles = append(regularProtoFiles, name)
 		}
 	}
-	var genProtoFiles []string
+
+	// Some of the generated files may have been consumed by other rules
+	consumedFileSet := make(map[string]bool)
+	for _, r := range args.OtherGen {
+		if r.Kind() != "proto_library" {
+			continue
+		}
+		for _, f := range r.AttrStrings("srcs") {
+			consumedFileSet[f] = true
+		}
+		if f := r.AttrString("src"); f != "" {
+			consumedFileSet[f] = true
+		}
+	}
+	var (
```

Comment:
`var genProtoFiles, genProtoFilesNotConsumed []string`

Also, please add a quick comment explaining these are separate so we don't generate empty rules for generated files that *are* consumed.